src/Tests/Cask.Tests/CaskSecretsTests.cs

```diff
@@ -257,45 +304,32 @@ private void TestEncodedMatchedDecoded(string encodedKey,
             ['_'] = 63
         };
 
-        char encodedSecretSizeChar = encodedTimestampSizesAndKindChars[0];
-        var encodedSecretSize = (SecretSize)base64UrlPrintableCharIndices[encodedSecretSizeChar];
-        Assert.Equal(secretSize, encodedSecretSize);
-
-        char encodedOptionalDataSizeChar = encodedTimestampSizesAndKindChars[1];
-        int optionalDataSizeInBytes = base64UrlPrintableCharIndices[encodedOptionalDataSizeChar] * 3;
-        Assert.Equal(paddedSecretSizeInBytes + 27 + optionalDataSizeInBytes, keyBytes.Length);
-
-        Range optionalDataRangeInBytes = sizesTimestampAndKindRangeInBytes.End..(sizesTimestampAndKindRangeInBytes.End.Value + optionalDataSizeInBytes);
-        Span<byte> optionalDataBytes = keyBytes[optionalDataRangeInBytes];
-        Assert.Equal(providerData, Base64Url.EncodeToString(optionalDataBytes));
-
-        int optionalDataSizeInChars = (optionalDataSizeInBytes / 3) * 4;
-        Range optionalDataRangeInChars = timestampSizesAndKindRangeInChars.End..(timestampSizesAndKindRangeInChars.End.Value + optionalDataSizeInChars);
-        string optionalDataInChars = encodedKey[optionalDataRangeInChars];
-        Assert.Equal(providerData, optionalDataInChars);
+        // Ensure encoded zero padding.
+        Assert.Equal('A', ppymdhmsChars[0]);
+        Assert.Equal('A', ppymdhmsChars[1]);
 
-        char encodedYearChar = encodedTimestampSizesAndKindChars[2];
-        char encodedMonthChar = encodedTimestampSizesAndKindChars[3];
-        char encodedDayChar = encodedTimestampSizesAndKindChars[4];
-        char encodedHourChar = encodedTimestampSizesAndKindChars[5];
-        char encodedMinuteChar = encodedTimestampSizesAndKindChars[6];
+        char encodedYearChar = ppymdhmsChars[2];
+        char encodedMonthChar = ppymdhmsChars[3];
+        char encodedDayChar = ppymdhmsChars[4];
+        char encodedHourChar = ppymdhmsChars[5];
+        char encodedMinuteChar = ppymdhmsChars[6];
+        char encodedSecondChar = ppymdhmsChars[7];
 
         int year = base64UrlPrintableCharIndices[encodedYearChar];
         int month = base64UrlPrintableCharIndices[encodedMonthChar];
         int day = base64UrlPrintableCharIndices[encodedDayChar];
         int hour = base64UrlPrintableCharIndices[encodedHourChar];
         int minute = base64UrlPrintableCharIndices[encodedMinuteChar];
+        int second = base64UrlPrintableCharIndices[encodedMinuteChar];
```

Comment:
```suggestion
        int second = base64UrlPrintableCharIndices[encodedSecondChar];
```

This is the first objective issue I've found on this PR.😊